File: VHBoomMenuButtonDemo/EaseVC.m
//
//  EaseVC.m
//  VHBoomMenuButton
//
//  Created by Nightonke on 2017/4/14.
//  Copyright © 2017 Nightonke. All rights reserved.
//

#import "EaseVC.h"
#import "BuilderManager.h"
#import <BoomMenuButton/BoomMenuButton.h>

@interface CustomTimeInterpolator : NSObject<VHTimeInterpolator>

@end

@implementation CustomTimeInterpolator

- (CGFloat)interpolation:(CGFloat)offset
{
    return offset < 0.5 ? offset : MIN(offset * 1.5, 1);
}

@end

@interface EaseVC ()

@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb1;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb2;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb3;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb4;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb5;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb6;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb7;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb8;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb9;

@end

@implementation EaseVC

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    [self initializeBMB:self.bmb1];
    [self initializeBMB:self.bmb2];
    [self initializeBMB:self.bmb3];
    [self initializeBMB:self.bmb4];
    [self initializeBMB:self.bmb5];
    [self initializeBMB:self.bmb6];
    [self initializeBMB:self.bmb7];
    [self initializeBMB:self.bmb8];
    [self initializeBMB:self.bmb9];
    
    self.bmb1.boomDuration = 1;
    self.bmb1.reboomDuration = 0.5;
    
    self.bmb2.boomEaseName = VHEaseOutExpo;
    self.bmb2.reboomEaseName = VHEaseInExpo;
    
    self.bmb3.boomEaseName = VHEaseOutBounce;
    self.bmb3.reboomEaseName = VHEaseOutBounce;
    
    self.bmb4.boomEaseName = VHEaseOutElastic;
    self.bmb4.reboomEaseName = VHEaseInElastic;
    self.bmb4.delay = 0;
    
    self.bmb5.boomEase = [CustomTimeInterpolator new];
    self.bmb5.reboomEase = [CustomTimeInterpolator new];
    self.bmb5.duration = 2;
    
    self.bmb6.boomEaseName = VHEaseInCirc;
    self.bmb6.reboomEaseName = VHEaseInCirc;
    
    self.bmb7.boomEaseName = VHEaseOutCirc;
    self.bmb7.reboomEaseName = VHEaseOutCirc;
    
    self.bmb8.boomMoveEaseName = VHEaseInBack;
    self.bmb8.boomRotateEaseName = VHEaseInBack;
    self.bmb8.boomScaleEaseName = VHEaseLinear;
    self.bmb8.reboomMoveEaseName = VHEaseInBack;
    self.bmb8.reboomRotateEaseName = VHEaseInBack;
    self.bmb8.reboomScaleEaseName = VHEaseInBack;
    
    self.bmb9.boomEaseName = VHEaseInOutQuad;
    self.bmb9.reboomEaseName = VHEaseInOutQuad;
    self.bmb9.delay = 0;
}

- (void)initializeBMB:(VHBoomMenuButton *)bmb
{
    bmb.buttonEnum = VHButtonSimpleCircle;
    bmb.piecePlaceEnum = VHPiecePlaceDOT_9_1;
    bmb.buttonPlaceEnum = VHButtonPlaceSC_9_1;
    for (int i = 0; i < bmb.pieceNumber; i++)
    {
        [bmb addBuilder:[BuilderManager simpleCircleButtonBuilder]];
    }
}

@end

File: VHBoomMenuButtonSwiftDemo/EaseVC.swift
//
//  EaseVC.swift
//  VHBoomMenuButtonSwiftDemo
//
//  Created by viktorhuang on 2017/5/4.
//  Copyright © 2017年 Nightonke. All rights reserved.
//

import UIKit
import BoomMenuButton

class EaseVC: BoomMenuButtonVC {

    @IBOutlet weak var bmb1: BoomMenuButton!
    @IBOutlet weak var bmb2: BoomMenuButton!
    @IBOutlet weak var bmb3: BoomMenuButton!
    @IBOutlet weak var bmb4: BoomMenuButton!
    @IBOutlet weak var bmb5: BoomMenuButton!
    @IBOutlet weak var bmb6: BoomMenuButton!
    @IBOutlet weak var bmb7: BoomMenuButton!
    @IBOutlet weak var bmb8: BoomMenuButton!
    @IBOutlet weak var bmb9: BoomMenuButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        initialize(boomMenuButton: bmb1)
        initialize(boomMenuButton: bmb2)
        initialize(boomMenuButton: bmb3)
        initialize(boomMenuButton: bmb4)
        initialize(boomMenuButton: bmb5)
        initialize(boomMenuButton: bmb6)
        initialize(boomMenuButton: bmb7)
        initialize(boomMenuButton: bmb8)
        initialize(boomMenuButton: bmb9)

        bmb1.boomDuration = 1
        bmb1.reboomDuration = 0.5
        
        bmb2.boomEaseName = Ease.outExpo
        bmb2.reboomEaseName = Ease.inExpo
        
        bmb3.boomEaseName = Ease.outBounce
        bmb3.reboomEaseName = Ease.outBounce
        
        bmb4.boomEaseName = Ease.outElastic
        bmb4.reboomEaseName = Ease.inElastic
        bmb4.delay = 0
        
        bmb5.boomEaseName = Ease.inSine
        bmb5.reboomEaseName = Ease.inSine
        
        bmb6.boomEaseName = Ease.inCirc
        bmb6.reboomEaseName = Ease.inCirc
        
        bmb7.boomEaseName = Ease.outCirc
        bmb7.reboomEaseName = Ease.outCirc
        
        bmb8.boomMoveEaseName = Ease.inBack
        bmb8.boomRotateEaseName = Ease.inBack
        bmb8.boomScaleEaseName = Ease.linear
        bmb8.reboomMoveEaseName = Ease.inBack
        bmb8.reboomRotateEaseName = Ease.inBack
        bmb8.reboomScaleEaseName = Ease.inBack
        
        bmb9.boomEaseName = Ease.inOutQuad
        bmb9.reboomEaseName = Ease.inOutQuad
        bmb9.delay = 0
    }

    func initialize(boomMenuButton bmb: BoomMenuButton) {
        bmb.buttonEnum = .simpleCircle
        bmb.piecePlaceEnum = .dot_9_1
        bmb.buttonPlaceEnum = .sc_9_1
        for _ in 0..<bmb.piecePlaceEnum.pieceNumber() {
            bmb.addBuilder(BuilderManager.simpleCircleButtonBuilder())
        }
    }

}
